File: rc4_flask_app/rc4.py
def rc4(key, data):
    S = list(range(256))
    j = 0
    out = []

    # Key Scheduling Algorithm (KSA)
    for i in range(256):
        j = (j + S[i] + key[i % len(key)]) % 256
        S[i], S[j] = S[j], S[i]

    # Pseudo-Random Generation Algorithm (PRGA)
    i = j = 0
    for char in data:
        i = (i + 1) % 256
        j = (j + S[i]) % 256
        S[i], S[j] = S[j], S[i]
        K = S[(S[i] + S[j]) % 256]
        out.append(chr(ord(char) ^ K))

    return ''.join(out)

def rc4_encrypt(key, plaintext):
    return rc4(key, plaintext)

def rc4_decrypt(key, ciphertext):
    return rc4(key, ciphertext)

File: RC4.java
public class RC4 {
    public static byte[] rc4(byte[] key, byte[] plaintext) {
        byte[] S = new byte[256];
        byte[] K = new byte[256];
        byte[] ciphertext = new byte[plaintext.length];

        int i, j = 0, t;

        // Key Scheduling Algorithm (KSA)
        for (i = 0; i < 256; i++) {
            S[i] = (byte) i;
            K[i] = key[i % key.length];
        }

        for (i = 0; i < 256; i++) {
            j = (j + S[i] + K[i]) & 0xFF;
            byte temp = S[i];
            S[i] = S[j];
            S[j] = temp;
        }

        // Pseudo-Random Generation Algorithm (PRGA)
        i = j = 0;
        for (int k = 0; k < plaintext.length; k++) {
            i = (i + 1) & 0xFF;
            j = (j + S[i]) & 0xFF;
            byte temp = S[i];
            S[i] = S[j];
            S[j] = temp;
            t = (S[i] + S[j]) & 0xFF;
            ciphertext[k] = (byte) (plaintext[k] ^ S[t]);
        }

        return ciphertext;
    }

    public static void main(String[] args) {
        String key = "clave";
        String plaintext = "texto plano";
        byte[] ciphertext = rc4(key.getBytes(), plaintext.getBytes());
        System.out.println("Cifrado: " + new String(ciphertext));
    }
}
